tooling/debugger/src/repl.rs

```diff
@@ -163,6 +262,23 @@ impl<'a, B: BlackBoxFunctionSolver> ReplDebugger<'a, B> {
         }
     }
 
+    fn restart_session(&mut self) {
+        let breakpoints: Vec<OpcodeLocation> =
+            self.context.iterate_breakpoints().cloned().collect();
+        self.context = DebugContext::new(
+            self.blackbox_solver,
+            self.circuit,
+            self.debug_artifact,
+            self.initial_witness.clone(),
```

Comment:
This is slightly strange as we're going to be restarting the execution with a pre-filled witness. It shouldn't cause any issues right now but may do if we do any inspection of the witness map.

We expose a function to return the witnesses needed to start execution so we should probably filter out any values which don't exist in this set. https://github.com/noir-lang/noir/blob/93d41e16570bbb0e53aaa9e31d40e7cf2f0eed63/acvm-repo/acir/src/circuit/mod.rs#L116-L119

Reply:
I'm not sure I follow here. The `initial_witness` in the `ReplDebugger` is always the original one (as passed from the command line arguments) and should not mutate over the debugging session.

I did run into the issue (in follow up work) of mutating the current witness map. I had to add a method to ACVM to do it since `witness_map()` returns an immutable reference. But in any case, the mutated witness map is not this one.

Reply:
Ah gotcha, we tend to abuse the term `initial_witness` elsewhere in the codebase to refer to a witness map which is being filled in so I assumed the same was happening here. This looks good to me in that case.